scripts/set_github_secrets.py

```diff
@@ -0,0 +1,461 @@
+from github.EnvironmentProtectionRuleReviewer import ReviewerParams
+from github.EnvironmentDeploymentBranchPolicy import (
+        EnvironmentDeploymentBranchPolicyParams,
+    )
+
+"""
+This script is used to set common secrets in all repositories
+When we add a new repository, add it to the repos list in the main function
+When we add a new environment, you need to do the following
+  - modify Secrets class to have roles for the new environment
+  - modify main function so all exports are retrieved for the new environment
+  - modify main function so roles are retrieved from exports for the new environment
+  - modify set_all_secrets function to set variables for the new environment
+
+You need a github token to run this and have aws config setup for each environment
+You can get a github token by first authenticating with github cli
+
+gh auth login
+
+and then get the token
+
+gh auth token
+
+This must be passed to this script as gh_auth_token param - eg
+
+GH_TOKEN=`gh auth token`
+poetry run python scripts/set_github_secrets.py --gh_auth_token $GH_TOKEN
+"""
+
+
+class Roles(TypedDict):
+    cloud_formation_deploy_role: str
+    cloud_formation_check_version_role: str
+    cloud_formation_prepare_changeset_role: str
+    CDK_pull_image_role: str
+    CDK_push_image_role: str
+    release_notes_execute_lambda_role: str
+    artillery_runner_role: str
+
+
+class Secrets(TypedDict):
+    regression_test_pem: str
+    automerge_pem: str
+    dev_roles: Roles
+    int_roles: Roles
+    prod_roles: Roles
+    qa_roles: Roles
+    ref_roles: Roles
+    recovery_roles: Roles
+    proxygen_prod_role: str
+    proxygen_ptl_role: str
+
+
+class GithubTeams(TypedDict):
+    eps_administrator_team: str
+    eps_testers_team: str
+    eps_team: str
+
+
+def get_named_export(all_exports: list, export_name: str, required: bool) -> str | None:
+    export_value = None
+
+    for export in all_exports:
+        if export['Name'] == export_name:
+            export_value = export['Value']
+            break
+
+    if required and export_value is None:
+        raise Exception(f"export {export_name} is required but not found")
+    return export_value
+
+
+def get_all_exports(profile_name: str) -> list:
+    print(f"Getting exports for profile {profile_name}")
+    session = boto3.Session(profile_name=profile_name)
+    # Create a CloudFormation client
+    cf_client = session.client('cloudformation')
+
+    # Get all exports
+    all_exports = []
+    next_token = None
+
+    while True:
+        if next_token:
+            response = cf_client.list_exports(NextToken=next_token)
+        else:
+            response = cf_client.list_exports()
+
+        all_exports.extend(response.get('Exports', []))
+
+        next_token = response.get('NextToken')
+        if not next_token:
+            break
+    return all_exports
+
+
+def get_role_exports(all_exports: list) -> Roles:
+    cloud_formation_deploy_role = get_named_export(all_exports,
```

Comment:
Similarly to the aws secrets, this should use the list of args, and process all this in a loop